### PR TITLE
Update docs to document volume subpaths (#559)

### PR DESCRIPTION
**Pull request by tylerslaton**
Issue #559

### CODE REVIEW

File: docs/docs/38-authoring/04-volumes.md [hunk 1]
```diff
@@ -35,6 +35,24 @@ volumes: {
 }
 ```
 
+## Volumes with subpaths
+
+When referencing volumes in containers, subpaths are supported. This feature allows you to have containers share a volume easily. You can utilize volume subpaths by adding a query parameter in the `dir`'s volume url like so.
```
Comment (cloudnautique) [inline docs/docs/38-authoring/04-volumes.md:40]: Can this start with a definition of what a sub-path is and when you would want to use it? Perhaps provide an example of the volume's filesystem and how it is used with the container here.

File: docs/docs/38-authoring/04-volumes.md [hunk 2]
```diff
@@ -35,6 +35,24 @@ volumes: {
...
+        image: "web"
+        // ...
+        dirs: {
+            "/data": "volume://my-data?subpath=data"
+        }
+    }
+}
+```
+
+If you'd like to see an example of subpaths in action you can take a look at our [Getting Started](../37-getting-started.md) guide. For more information on the concept of subpaths in general, you can checkout the official [Kubernetes documentation](https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath) on them.
```
Comment (cloudnautique) [inline docs/docs/38-authoring/04-volumes.md:54]: Can we provide enough info here to prevent a user from having to go to K8s docs?

Comment (tylerslaton) [reply docs/docs/38-authoring/04-volumes.md:54]: Yep! Good call. Let me update this to be more descriptive on subpaths.

File: docs/docs/38-authoring/04-volumes.md (hunk 1 above)
Comment (jsilverio22) [reply docs/docs/38-authoring/04-volumes.md:40]: +1

Comment (tylerslaton) [reply docs/docs/38-authoring/04-volumes.md]: Alright, did an uplift to make this more descriptive. Tell me what you think!

File: docs/docs/38-authoring/04-volumes.md [hunk 3]
```diff
@@ -35,6 +35,69 @@ volumes: {
...
+
+```
+example-volume
+|- data-1
+   |- nested-data-1
+   |- nested-data-2
+|- data-2
+```
+
+In this example, we want create a mount for the content found in `data-1` to a container. Without subpaths, our Acornfile would look something like this.
```
Comment (cjellick) [inline docs/docs/38-authoring/04-volumes.md:50]:
```suggestion
In this example, we want to create a mount for the content found in `data-1` to a container. Without subpaths, our Acornfile would look something like this.
```

**Review by cloudnautique — APPROVED** [review]

**Review by cjellick — APPROVED** [review]

**Pull request merged**